src/main/java/org/myrobotlab/service/interfaces/ServoControl.java

```diff
@@ -120,150 +168,249 @@
+   * by an encoder.
    * 
    * A possible better solution might be to use ServoEvent(s) to get the
    * position through time which a controller with speed control can provide, so
    * as the servo is told incrementally where to go - it sends that command back
    * as an event which sets the "current" position.
    * 
-   * @return the current position as a double
+   * This is a read only current position reported from the ServoControl. If a
+   * mapper calculation is supplied to generate output - and the ServoController
+   * uses it, and the ServoController sends back current position it "must" be
+   * in the form of the original input range
+   *
+   * set by feedback encoders if available - this is typically updated during
+   * movement - a read-only value
+   * 
+   * @return the current position as a Double
    */
-  double getPos();
+  Double getPos();
 
   /**
-   * the calculated mapper output for the servo - this is <b> ALWAYS ALWAYS in
-   * DEGREES !</b> because this method is used by the controller - and the
-   * controller needs a stable method &amp; a stable unit FIXME - not sure if
-   * this is a good thing to expose
+   * @return the current rest position value
+   */
+  Double getRest();
+
+  /**
+   * Return current speed if set - if speed/speed control is not being use it is
+   * null.
+   * 
+   * @return
+   */
+  Double getSpeed();
+
+  /**
+   * This value is for the ServoController to consume.
+   * 
+   * The calculated mapper output for the servo - this is <b> ALWAYS ALWAYS in
+   * DEGREES !</b> because this method is used by the servo controller - and the
+   * controller needs a stable method &amp; a stable unit
+   * 
+   * The is the output of the mapper, the servo controller may have its own
+   * global mapper, this one is unique for a specific Servo
+   * 
+   * This is position the mapper returns after the calculation of mapper
    * 
    * @return the target output position
    */
-  double getTargetOutput();
+  Double getTargetOutput();
 
-  double getMaxVelocity();
+  /**
+   * If currently attached to a controller
+   * 
+   * @return
+   */
+  Boolean isAttached();
 
-  double getVelocity();
+  /**
+   * is the servo currently sending pwm position control
+   * @return
+   */
+  Boolean isEnabled();
 
   /**
-   * set the pin of the servo this does not 'attach' energize the pin only set
-   * the pin value
+   * Returns true if mapper is inverted
    * 
-   * @param pin
-   *          the pin number for the servo
+   * @return
    */
-  void setPin(int pin);
+  Boolean isInverted();
 
   /**
-   * @return the current acceleration value
+   * This sets the servo's mapper explicitly
+   * 
+   * @param minX
+   *          - min input
+   * @param maxX
+   *          - max input
+   * @param minY
+   *          - min output
+   * @param maxY
+   *          - max output
    */
-  double getAcceleration();
+  void map(Double minX, Double maxX, Double minY, Double maxY);

```

Comment:
I would much prefer just see a simple set /get Mapper(Mapper m) in the interface rather than having to pass 4 doubles.